### ISSUE TITLE
[Bug]: Incorrect crop after constrained layout with equal aspect ratio and bbox_inches = tight

### ISSUE THREAD

**Issue by konradmarx** [matplotlib/matplotlib#24133]
### Bug summary

When setting `rcParams['figure.constrained_layout.use'] = True` and `ax.set_aspect('equal')`, and saving with `bbox_inches = 'tight'`, the resulting figure is cropped incorrectly.

### Code for reproduction

```python
import matplotlib as mpl
import matplotlib.pyplot as plt

mpl.rcParams['figure.constrained_layout.use'] = True

fig, ax = plt.subplots()

ax.plot([0, 1])
ax.set_xlim(0, 1)
ax.set_aspect('equal')

fig.savefig('test.png', bbox_inches = 'tight')
```

### Actual outcome

```
MatplotlibDeprecationWarning: The resize_event function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use callbacks.process('resize_event', ResizeEvent(...)) instead.
  fig, ax = plt.subplots()
```
![test](https://user-images.githubusercontent.com/64837956/194764228-8b349e42-0ddb-48a8-ba59-a1827cb3f4d3.png)

### Expected outcome

If I leave out `mpl.rcParams['figure.constrained_layout.use'] = True`, but call `subplots` with `layout = 'constrained'`, the saved figure is as expected:
![test](https://user-images.githubusercontent.com/64837956/194764346-8b12b2f5-44f8-4ba5-8440-2fd5d2581cb1.png)

(The deprecation warning is probably the same is in [#23921](https://github.com/matplotlib/matplotlib/issues/23921), and appears in both cases.)

### Operating system

macOS Monterey

### Matplotlib Version

3.6.0

### Matplotlib Backend

macOSX

### Python version

3.10.6

### Installation

pip

**Comment by Anupam-USP** [comment]
Hey @konradmarx, if this issue is still open can you please help me on getting started with this?

**Comment by konradmarx** [comment]
Hi, thanks for the answer!
The issue is still open and also occurs in version 3.6.1 (only the deprecation warning is gone).
However, if I downgrade to 3.5.2, the issue doesn't occur.

**Comment by jklymak** [comment]
Thanks for the report and sorry this broke.  It should be fixed in 3.6.2

**Comment by konradmarx** [comment]
Thanks for fixing it!
